src/buildstream/buildelement.py

```diff
@@ -227,9 +229,19 @@ def configure_dependencies(self, dependencies):
         for dep in dependencies:
             # Determine the location to stage each element, default is "/"
             location = "/"
+
             if dep.config:
-                dep.config.validate_keys(["location"])
-                location = dep.config.get_str("location")
+                dep.config.validate_keys(["digest-environment", "location"])
+
+                if "location" in dep.config:
+                    location = dep.config.get_str("location")
+
+                if "digest-environment" in dep.config:
```

Comment:
Instead of doing a double dictionary lookup, we should use the default `None` to `MappingNode.get_str()`.

```python
   var_name = dep.config.get_str("digest-environment", None)
   if var_name is not None:
       element_list = blablabla
```

Reply:
Done.